addons/auth_oauth/models/auth_oauth.py

```diff
@@ -21,3 +23,12 @@ class AuthOAuthProvider(models.Model):
     css_class = fields.Char(string='CSS class', default='fa fa-fw fa-sign-in text-primary')
     body = fields.Char(required=True, help='Link text in Login Dialog', translate=True)
     sequence = fields.Integer(default=10)
+
+    @api.model
+    def _neutralize(self):
+        super()._neutralize()
+        provider_ids = self.search([('enabled', '=', True)])
+        provider_ids.update({'enabled': False})
+        for provider in provider_ids:
+            _logger.warning('Oauth provider `%s` neutralized', provider.name)
+        assert self.search([('enabled', '=', True)]) == self.env['auth.oauth.provider'], "Not all oauth providers were neutralized !"
```

Comment:
Maybe the log above should be an INFO, and the assert be a WARNING or ERROR?

Reply:
I agree with the INFO.
But about the assert, isn't it better to have a complete failure ?
Use case: someone runs the neutralize process with the CLI and does not see a single warning passing in the logs.
He expects to have a neutralized database ... but it's not the case.

Reply:
How does it integrate with SH or Saas database?
Does it prevent to creation of the duplicate?

Or do we get a duplicate with no neutralization at all?
I think I prefer to have most of the neutralization done in all case even if a single one fails.
What about a warning on the App Switcher in case of failure of any neutralization with details about it? (if possible)